request/extractor.go

```diff
@@ -79,3 +80,18 @@ func (e *PostExtractionFilter) ExtractToken(req *http.Request) (string, error) {
 		return "", err
 	}
 }
+
+// BearerExtractor extracts a token from the Authorization header.
+// The header is expected to match the format "Bearer XX", where "XX" is the
+// JWT token.
+type BearerExtractor struct{}
+
+func (e BearerExtractor) ExtractToken(req *http.Request) (string, error) {
+	tokenHeader := req.Header.Get("Authorization")
+	// The usual convention is for "Bearer" to be title-cased. However, there's no
+	// strict rule around this, and it's best to follow the robustness principle here.
+	if tokenHeader == "" || !strings.HasPrefix(strings.ToLower(tokenHeader), "bearer ") {
+		return "", ErrNoTokenInRequest
+	}
+	return strings.TrimPrefix(tokenHeader, "Bearer "), nil
```

Comment:
```suggestion
	return strings.TrimPrefix(strings.ToLower(tokenHeader), "bearer "), nil
```

Reply:
This will convert the token into lowercase. Are tokens themselves case-insensitive too?

Reply:
Oops. No they are not

Reply:
Updated PR to avoid messing up case of the token.

Reply:
And added tests for that.